src/client/index.ts

```diff
@@ -4,9 +4,15 @@
  */
 
 import TaskClient from './client';
-import CosmosDbClient from './cosmos';
+import CosmosDbClient, { ChainedTokenCredential } from './cosmos';
```

Comment:
the same comment applies about not exporting this type, but also why export it from the cosmos client just to reimport/reexport it here? seems clearer to do the import/re-export from identity directly here (which to be clear i still don't recommend for the reasons in my other comment)

Reply:
actually sorry i realized there's another level of import/export above this. _that's_ where i would do it